### PR TITLE
Added Junit and IT test cases and endpoint to fetch all data from mongo db

### PR DESCRIPTION
**Pull request by Arundhathy-k**
Added Junit and IT test cases for NewsService module. All test cases are passed, and coverage is more than 90%. 
Added endpoint to fetch all data from mongo db.

### CODE REVIEW

File: mongo-dao/src/main/java/com/kovan/dto/NewsDto.java [hunk 1]
```diff
@@ -28,6 +28,7 @@ public class NewsDto {
 
     private   List<Article> articles;
 
+
```
Comment (dhanyesh) [inline mongo-dao/src/main/java/com/kovan/dto/NewsDto.java:25]: why do we need this space

File: newsService/pom.xml [hunk 2]
```diff
@@ -41,6 +41,24 @@
             <version>0.0.1-SNAPSHOT</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
```
Comment (dhanyesh) [inline newsService/pom.xml:44]: why do we need this dependency here.(jakson)

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java [hunk 3]
```diff
@@ -7,6 +7,8 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
```
Comment (dhanyesh) [inline newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java:10]: remove extra space

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java [hunk 4]
```diff
@@ -15,10 +17,15 @@ public class NewsServiceController {
...
     public ResponseEntity<NewsDto> fetchAndSaveTopHeadlines() {
 
         return new ResponseEntity<>(newsService.getTopHeadlines(), HttpStatus.OK);
     }
 
+    @GetMapping("/fetchAllNews")
+    public ResponseEntity<List<NewsDto>> fetchAllNewsFromDB(){
+        return new ResponseEntity<>(newsService.getAllData(),HttpStatus.OK);
+    }
+
```
Comment (dhanyesh) [inline newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java:27]: remove extra space

File: newsService/src/main/java/com/kovan/app/service/NewsService.java [hunk 5]
```diff
@@ -66,7 +66,7 @@ public NewsDto getTopHeadlines() {
         } while (pageSize <= newsDto.getTotalResults());
 
         LocalDate dateOnly = LocalDate.parse(
-                articles.get(0).getPublishedAt(),
+                articles.getFirst().getPublishedAt(),
```
Comment (dhanyesh) [inline newsService/src/main/java/com/kovan/app/service/NewsService.java:70]: Good you learned getFirst() method. appreciate it!

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+public class NewsServiceControllerIT {
+
+    private final NewsService newsService;
+    private final NewsRepository newsRepository;
+
+    @Autowired
```
Comment (dhanyesh) [inline newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java:20]: Is this really needed ?

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Test
+    public void testFetchAllNewsFromDB() {
+        NewsDto newsDto = newsService.getTopHeadlines();
+        List<NewsDto> newsList = newsService.getAllData();
+        System.out.println(newsList.size());
+
+        assertThat(newsList).isNotNull();
+        assertThat(newsList.size()).isGreaterThan(0);
+        cleanup();
```
Comment (dhanyesh) [inline newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java:46]: check assert after that you have to validate datas are getting flushed or not

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        assertThat(newsDto.getTotalResults()).isGreaterThan(0);
+        assertThat(newsDto.getArticles()).isNotNull();
+        assertThat(newsDto.getArticles().size()).isGreaterThan(0);
+    }
+
+    @Test
+    public void testFetchAllNewsFromDB() {
+        NewsDto newsDto = newsService.getTopHeadlines();
+        List<NewsDto> newsList = newsService.getAllData();
+        System.out.println(newsList.size());
```
Comment (dhanyesh) [inline newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java]: Don't need println I asked you to add sysout only for your testing.

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+        assertThat(newsList).isNotNull();
+        assertThat(newsList.size()).isGreaterThan(0);
+        cleanup();
+    }
+
+    private void cleanup(){
+        newsRepository.deleteAll();
+    }
+
```
Comment (dhanyesh) [inline newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java:56]: remove extra space

File: newsService/src/test/java/com/kovan/unit/NewsServiceTests.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Mock
+    private NewsRepositoryService newsRepositoryService;
+
+    @Mock
+    private NewsRepository newsRepository;
+
+    @InjectMocks
+    private NewsService newsService;
+
```
Comment (dhanyesh) [inline newsService/src/test/java/com/kovan/unit/NewsServiceTests.java:41]: remove extra space

**Review by dhanyesh — CHANGES_REQUESTED** [review]

File: mongo-dao/src/main/java/com/kovan/dto/NewsDto.java (hunk 1 above)
Comment (Arundhathy-k) [reply mongo-dao/src/main/java/com/kovan/dto/NewsDto.java:25]: Removed the extra space.

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java (hunk 3 above)
Comment (Arundhathy-k) [reply newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java:10]: Removed the extra space.

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java (hunk 4 above)
Comment (Arundhathy-k) [reply newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java:27]: Removed extra space.

File: newsService/src/main/java/com/kovan/app/service/NewsService.java (hunk 5 above)
Comment (Arundhathy-k) [reply newsService/src/main/java/com/kovan/app/service/NewsService.java:70]: Thanks for the appreciation.

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java (hunk 8 above)
Comment (Arundhathy-k) [reply newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java]: Removed println.

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java (hunk 9 above)
Comment (Arundhathy-k) [reply newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java:56]: Removed extra spaces.

File: newsService/src/test/java/com/kovan/unit/NewsServiceTests.java (hunk 10 above)
Comment (Arundhathy-k) [reply newsService/src/test/java/com/kovan/unit/NewsServiceTests.java:41]: Removed extra space.

File: newsService/pom.xml (hunk 2 above)
Comment (Arundhathy-k) [reply newsService/pom.xml:44]: Removed dependency.

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java (hunk 6 above)
Comment (Arundhathy-k) [reply newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java:20]: Yes. Junit 5 is not supporting constructor injection. So @Autowired annotation is needed.

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java (hunk 7 above)
Comment (Arundhathy-k) [reply newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java:46]: assert added .

File: mongo-dao/src/main/java/com/kovan/dto/NewsDto.java [hunk 11]
```diff
@@ -47,8 +42,8 @@ public static class Article {
         @AllArgsConstructor
         @NoArgsConstructor
         public static class Source {
-          private String id;
-          private String name;
+            private String id;
+            private String name;
```
Comment (dhanyesh) [inline mongo-dao/src/main/java/com/kovan/dto/NewsDto.java:47]: remove extra space

File: newsService/pom.xml [hunk 12]
```diff
@@ -41,6 +41,14 @@
...
+            <groupId>org.springframework.boot</groupId>
```
Comment (dhanyesh) [inline newsService/pom.xml:45]: check this dependency is needed or not

File: mongo-dao/src/main/java/com/kovan/dto/NewsDto.java [hunk 13]
```diff
@@ -1,15 +1,9 @@
...
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
-import org.springframework.data.annotation.Id;
-
-import java.time.LocalDate;
-import java.time.LocalDateTime;
 import java.util.List;
 
 @Data
```
Comment (dhanyesh) [inline mongo-dao/src/main/java/com/kovan/dto/NewsDto.java:9]: where is @Builder annotation?

**Review by dhanyesh — CHANGES_REQUESTED** [review]

**Review by dhanyesh — CHANGES_REQUESTED** [review]
Where is new branch for Junit ticket ?

File: mongo-dao/src/main/java/com/kovan/dto/NewsDto.java (hunk 11 above)
Comment (Arundhathy-k) [reply mongo-dao/src/main/java/com/kovan/dto/NewsDto.java:47]: removed extra space.

File: mongo-dao/src/main/java/com/kovan/dto/NewsDto.java (hunk 13 above)
Comment (Arundhathy-k) [reply mongo-dao/src/main/java/com/kovan/dto/NewsDto.java:9]: @builder annotation already exists.

File: newsService/pom.xml (hunk 12 above)
Comment (Arundhathy-k) [reply newsService/pom.xml:45]: Removed that dependency.

**Comment by Arundhathy-k** [comment]
Forgot to create new branch. Hence closing this pull request.

**Comment by Arundhathy-k** [comment]
Forgot to create new branch. Hence closing this pull request.

**Pull request closed without merging**